Twittimation/RandomLikeAutomaton.cs

```diff
@@ -0,0 +1,104 @@
+
+        public RandomLikeAutomaton(IStored<RandomLikeAutomatonData> data, IStored<Tasks> tasks, Like like, IStored<Credentials> credentials)
+        {
+            _data = data;
+            _tasks = tasks;
+            _like = like;
+            _credentials = credentials;
+        }
+
+        public void Update()
+        {
+            if(_updatesToSkip == 0)
+            {
+                _updatesToSkip = 11;
+                var credentials = _credentials.Get();
+                if (credentials.AreValid)
+                {
+                    var now = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+                    var random = new Random();
+                    try
+                    {
+                        var oauthClient = new OAuthClient("https://api.twitter.com/1.1/",
+                            credentials.ConsumerKey, credentials.ConsumerKeySecret, credentials.AccessToken, credentials.AccessTokenSecret);
+                        var followers = JObject.Parse(oauthClient.SendRequest("GET", "friends/ids.json", new JsonData()).Result);
+                        if (followers.ContainsKey("errors"))
+                        {
+                            Console.Error.WriteLine("Failed to get followers!");
+                            return;
+                        }
+                        var tasks = _tasks.Get();
+                        _data.Update(d => {
+                            if (d.LastUpdatedTimestamp < now - now % (SecondsPerHour * 24))
+                                d.LikesGivenPerFollowee = d.LikesGivenPerFollowee.ToDictionary(kvp => kvp.Key, kvp => 0);
+                            foreach(var id in followers.GetValue("ids").ToObject<long[]>())
+                            {
+                                var tweets = JsonConvert.DeserializeObject<List<TwitterTweet>>(
+                                    oauthClient.SendRequest("GET", "statuses/user_timeline.json",
+                                        new Dictionary<string, string>() {
+                                            { "user_id", id.ToString() },
+                                            { "count", "50" },
+                                            { "include_rts", "false" },
+                                            { "trim_user", "true" },
+                                            { "exclude_replies", "true" }
+                                        },
+                                        new JsonData())
+                                    .Result);
+                                if (d.LikesGivenPerFollowee.ContainsKey(id))
+                                    tweets = tweets.Where(t => t.CreatedAt.ToUnixTimeSeconds() > d.LastUpdatedTimestamp).ToList();
+                                else
+                                    d.LikesGivenPerFollowee.Add(id, 0);
+                                tweets = tweets.Where(t => t.CreatedAt.ToUnixTimeSeconds() > (now - SecondsPerHour * 24 * 30)).ToList();
```

Comment:
Would be better to create a projection, instead of mutating the variable.

Reply:
What do you mean?

Reply:
Like `var newTweets = tweets.Where(...).ToList()`

Reply:
It's bad practice to re-assign a non-counter variable.

Reply:
What is the alternative you are trying to propose?

Reply:
There is literally sample code with the proposal...